Edit proposal

alexisljn opened the pull request.

alexisljn committed:
Push To Work On Another Computer

alexisljn committed:
Initialize Edit Proposal Form With Existing Content

Install of js library turndown
script written in views

alexisljn commented on config/web.php:
```diff
@@ -64,6 +64,7 @@
             'showScriptName' => false,
             'rules' => [
                 'proposal/my-proposals/<id:\d+>' => 'proposal/my-proposals',
+                'proposal/edit-proposal/<id:\d+>' => 'proposal/edit-proposal',
```
I'm going to change it in further commit

alexisljn commented on views/proposal/my-proposal.php:
```diff
@@ -65,4 +83,24 @@
+            const proposalContent = new TurndownService({
+                headingStyle: 'atx',
+                bulletListMarker: '-' }).turndown($('#original-content').html());
+            console.log(proposalContent);
```
Will be removed of course ;)

alexisljn committed:
Start Edit Save

And removal of useless lib.

alexisljn committed:
Complete Form Of Editing Proposal

File logic is needed.

alexisljn committed:
Creation of File Content History Entity

For chronological Stream

alexisljn committed:
Recreating Model And Change Import SQL File

proposal_file_history
import SQL

alexisljn committed:
Finalize Edit Proposal Logic

Implementing Proposal File History

alexisljn committed:
Add PHPDoc To Methods

For better understanding.

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+    {
+        $editedProposal = Proposal::findOne(['id' => Yii::$app->request->get()]);
+        $lastProposalContent = $editedProposal->proposalContentHistories[
+        count($editedProposal->proposalContentHistories)-1];
```
Please indent this line

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+        $postRequest = Yii::$app->request->post();
+
+        $model->title = $postRequest['ManageProposalForm']['title'];
+        $model->content = $postRequest['ManageProposalForm']['content'];
```
Those two lines are useless: The $model->load function already does this

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+        if ($model->load(Yii::$app->request->post()) && $model->validate()) {
+            $transaction = Yii::$app->db->beginTransaction();
+            try {
+                $this->saveEditedProposal($postRequest['ManageProposalForm']['title'], $editedProposal);
```
Why don't you use $model->title instead of $postRequest['ManageProposalForm']['title'] ?

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+            try {
+                $this->saveEditedProposal($postRequest['ManageProposalForm']['title'], $editedProposal);
+
+                if ($postRequest['ManageProposalForm']['content'] != $lastProposalContent->content) {
```
Idem, you should use $model->content

GuenoleCHERUBIN commented on controllers/ProposalController.php:
```diff
@@ -90,7 +100,7 @@ private function checkIfUserIsOwnerOfProposal(int $submitterId)
         }
 
-        return $unauthorizedException();
+        throw new $unauthorizedException();
```
Why $unauthorizedException ?

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+                $this->saveEditedProposal($postRequest['ManageProposalForm']['title'], $editedProposal);
+
+                if ($postRequest['ManageProposalForm']['content'] != $lastProposalContent->content) {
+                    $this->saveProposalContent($postRequest['ManageProposalForm']['content'], $editedProposal, false);
```
Idem

EmileCalixte commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+     */
+    private function saveEditedFile($movedFilename, Proposal $proposal)
+    {
+        $file = $proposal->files[0];
```
We have to edit the database model : a proposal have maximum 1 file, so the field proposal_id in the file table must be unique

EmileCalixte replied:
After that, you will have to use $file = $proposal->file

GuenoleCHERUBIN commented on controllers/ProposalController.php:
```diff
@@ -418,4 +439,126 @@ private function saveProposalRelatedFile(string $movedFilename, Proposal $propos
+            $proposalFileHistory->date = $proposal->date;
+        }
+        $proposalFileHistory->date = Util::getDateTimeFormattedForDatabase(new \DateTime());
+
```
Missing an else ?

GuenoleCHERUBIN commented on sql/import.sql:
```diff
@@ -19,7 +19,7 @@ SET time_zone = "+00:00";
 
 --
--- Base de données : `collabmedia`
+-- Base de données :  `collabmedia`
```
To much spaces ?

GuenoleCHERUBIN commented on views/proposal/my-proposal.php:
```diff
@@ -2,24 +2,46 @@
-<p>Created at <?= $selectedProposal->date ?></p>
+
+
+
```
Why so much return ?

alexisljn committed:
Correct For PR Approval

- Modification of Proposal and File Models
- Minor corrections in view and controller
- Overwriting of .sql used as fixtures.

alexisljn committed:
Merge branch 'master' into edit-proposal
